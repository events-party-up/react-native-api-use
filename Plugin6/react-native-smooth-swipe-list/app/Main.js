import React from 'react';
import { View, StyleSheet } from 'react-native';

import { TodoScene } from './scenes';

const Main = () => (
    <View style={styles.container}>
        <TodoScene/>
    </View>
);


const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'flex-start',
        marginTop: 25
    }
});


export default Main;