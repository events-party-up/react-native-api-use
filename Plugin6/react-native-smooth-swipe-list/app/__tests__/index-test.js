import 'react-native';
import React from 'react';
import Main from '../index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders Main correctly', () => {
    const tree = renderer.create(
        <Main />
    );
    expect(tree).toMatchSnapshot();
});
