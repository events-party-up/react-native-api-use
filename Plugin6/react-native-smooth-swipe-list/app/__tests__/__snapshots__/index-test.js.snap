exports[`test renders Main correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "left": 0,
      "marginTop": 25,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "stretch",
        "color": "rgb(10, 110, 150)",
        "fontSize": 22,
        "paddingHorizontal": 25,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }>
    Example List
  </Text>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      }
    }>
    <ScrollView
      dataSource={
        ListViewDataSource {
          "items": 3,
        }
      }
      enableEmptySections={true}
      gestureTensionParams={
        Object {
          "length": 1,
        }
      }
      initialListSize={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={1000}
      onKeyboardDidHide={undefined}
      onKeyboardDidShow={undefined}
      onKeyboardWillHide={undefined}
      onKeyboardWillShow={undefined}
      onLayout={[Function]}
      onScroll={[Function]}
      onScrollStateChange={[Function]}
      pageSize={1}
      removeClippedSubviews={true}
      renderFooter={[Function]}
      renderRow={[Function]}
      rowData={
        Array [
          Object {
            "key": 1,
            "leftSubView": <FullWidthSubItem />,
            "leftSubViewOptions": Object {
              "closeOnClick": false,
              "fullWidth": true,
            },
            "rightSubView": <ListItemButton
              color="primary"
              onPress={[Function]}
              text="Archive" />,
            "rowView": <ListItem
              onPress={[Function]}
              title="List Item One" />,
          },
          Object {
            "key": 2,
            "leftSubView": <FullWidthSubItem />,
            "leftSubViewOptions": Object {
              "closeOnClick": false,
              "fullWidth": true,
            },
            "rightSubView": <ListItemButton
              color="primary"
              onPress={[Function]}
              text="Archive" />,
            "rowView": <ListItem
              onPress={[Function]}
              title="List Item Two" />,
          },
          Object {
            "key": 3,
            "leftSubView": <FullWidthSubItem />,
            "leftSubViewOptions": Object {
              "closeOnClick": false,
              "fullWidth": true,
            },
            "rightSubView": <ListItemButton
              color="primary"
              onPress={[Function]}
              text="Archive" />,
            "rowView": <ListItem
              onPress={[Function]}
              title="List Item Three" />,
          },
        ]
      }
      rowStyle={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#eee",
          "height": 55,
        }
      }
      scrollEnabled={true}
      scrollEventThrottle={50}
      scrollRenderAheadDistance={1000}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "rgb(111, 111, 111)",
          },
          Object {
            "backgroundColor": "rgb(247, 247, 247)",
          },
        ]
      }>
      <View
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#eee",
                "height": 55,
              },
              undefined,
            ],
          ]
        }>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignSelf": "stretch",
                "left": 0,
                "top": 0,
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderBottomColor": "#eee",
                      "borderBottomWidth": 1,
                      "height": 55,
                      "justifyContent": "center",
                      "paddingHorizontal": 25,
                    },
                    undefined,
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#444",
                      },
                      undefined,
                    ]
                  }>
                  List Item One
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#eee",
                "height": 55,
              },
              undefined,
            ],
          ]
        }>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignSelf": "stretch",
                "left": 0,
                "top": 0,
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderBottomColor": "#eee",
                      "borderBottomWidth": 1,
                      "height": 55,
                      "justifyContent": "center",
                      "paddingHorizontal": 25,
                    },
                    undefined,
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#444",
                      },
                      undefined,
                    ]
                  }>
                  List Item Two
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#eee",
                "height": 55,
              },
              undefined,
            ],
          ]
        }>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignSelf": "stretch",
                "left": 0,
                "top": 0,
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderBottomColor": "#eee",
                      "borderBottomWidth": 1,
                      "height": 55,
                      "justifyContent": "center",
                      "paddingHorizontal": 25,
                    },
                    undefined,
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#444",
                      },
                      undefined,
                    ]
                  }>
                  List Item Three
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}>
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "height": 55,
                "justifyContent": "center",
                "paddingHorizontal": 25,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(44, 217, 222)",
              },
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#444",
                },
                Object {
                  "color": "rgb(250, 250, 250)",
                  "fontWeight": "bold",
                },
              ]
            }>
            Load More
          </Text>
        </View>
      </View>
    </ScrollView>
  </View>
</View>
`;
